crypto/merkletrie/committer_test.go

```diff
@@ -33,9 +34,7 @@ func (mc *InMemoryCommitter) Duplicate(flat bool) (out *InMemoryCommitter) {
 		if flat {
 			out.memStore[k] = v
 		} else {
-			bytes := make([]byte, len(v))
-			copy(bytes[:], v[:])
-			out.memStore[k] = bytes
+			out.memStore[k] = slices.Clone(v)
```

Comment:
Good question.  I think they reduce to the exact same code. Part of me says we should use the specific call, because it seems like that's the point of `bytes.Clone` existing. But another part thinks the existence of `bytes.Clone` is just a historic artifact from before generics, so why write different things for different types?

What do you (or anyone else) think?

Reply:
I can't think of a reason to have a preference for one over the other, but I wanted to have the discussion in case someone else knows of one.

FWIW `bytes.Clone` was introduced in 1.20, so definitely not pre-generics.

Reply:
I just read a bit about it. Nothing conclusive, but I get the impression it only went in because strings.Clone exists, and they have some sort of pathological desire to make the API surface of those identical (even going through the trouble of adding strings.Compare and explicitly making it slow and recommending against use!)

So I'm going to say I lean toward slices.Clone everywhere. But I don't feel strongly.